mera/utils.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+import pandas as pd
+
+
+def mask_too_few_records(
+    residual_df: pd.DataFrame,
+    column_name1: Optional[str] = "event_id",
+    column_name2: Optional[str] = "stat_id",
+    min_num_records_per_column1: Optional[int] = 3,
+    min_num_records_per_column2: Optional[int] = 3,
+    verbose: Optional[bool] = False,
+) -> pd.DataFrame:
+    """
+    Mask records if there are too few records per event or per station to be useful.
+
+    Parameters
+    ----------
+    residual_df : pd.DataFrame
+        The residual DataFrame to mask.
+    column_name1 : str, default = 'event_id'
+        The first column to consider. The default value is 'event_id', which contains the event ids.
+    column_name2 : str, default = 'stat_id'
+        The second column to consider. The default value is 'stat_id', which contains the station ids.
+    min_num_records_per_column1 : int, default = 3
+        The minimum number of records required in `column_name1` to keep the records.
+    min_num_records_per_column2 : int, default = 3
+        The minimum number of records required in `column_name2` to keep the records.
+    verbose: bool, default = False
+        If True, print the station and event ids that were masked and the number of records they were associated with.
+
+    Returns
+    -------
+    final_mask2D : pd.DataFrame
```

Comment:
Don't need final in the name here, also don't need 2D imo. I would have something like:
```
Returns
--------
mask: pd.DataFrame
    Mask dataframe with the same shape as the given residual dataframe.
```